### PR TITLE
Feature/compiling+market functions

### PR DESCRIPTION
**Pull request by LukasDeco**
- Refactor RPC openbookMarkets client
- Add deposit and withdrawal to proposal
- types compiling

### CODE REVIEW

File: lib/client/rpc/openbookMarkets.ts [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+    market: OBMarket;
+  }): TwapPlaceOrderArgs | undefined {
+    let priceLots = market.priceLotsToUi(price);
+    const _priceLots = market.priceLotsToUi(price);
+    const maxBaseLots = market.baseLotsToUi(amount);
+    let maxQuoteLotsIncludingFees = market.quoteLotsToUi(
+      priceLots * maxBaseLots
+    );
+
+    if (!isLimitOrder) {
```
Comment (R-K-H) [inline lib/client/rpc/openbookMarkets.ts:306]: we want to use placeTakeOrder if we're not dealing with a limit.

File: lib/client/rpc/balances.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+exports.FutarchyRPCBalancesClient = void 0;
+const spl_token_1 = require("@solana/spl-token");
+const web3_js_1 = require("@solana/web3.js");
+class FutarchyRPCBalancesClient {
+    constructor(rpcProvider) {
+        this.rpcProvider = rpcProvider;
+    }
+    async fetchMainTokenWalletBalances(dao, ownerWallet) {
```
Comment (R-K-H) [inline lib/client/rpc/balances.js:10]: Could we generalize this a bit more to accept single pub key or list of pub keys (or just create a new function).

File: lib/client/rpc/dao.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+exports.FutarchyRPCDaoClient = void 0;
+const tokens_1 = require("../../tokens");
+class FutarchyRPCDaoClient {
+    constructor(rpcProvider, programVersion, autocratProgram) {
+        this.rpcProvider = rpcProvider;
+        this.programVersion = programVersion;
+        this.autocratProgram = autocratProgram;
+    }
```
Comment (R-K-H) [inline lib/client/rpc/dao.js:10]: All autocrats (doesn't have to be this PR)

File: lib/client/rpc/openbookMarkets.ts (hunk 1 above)
Comment (LukasDeco) [reply lib/client/rpc/openbookMarkets.ts]: this is handled

File: lib/client/rpc/balances.js (hunk 2 above)
Comment (LukasDeco) [reply lib/client/rpc/balances.js:10]: Added a TODO in the comments about this per our discussion.

**Review by R-K-H — APPROVED** [review]
LGTM We'll test out the place take order as a TODO: Now granted, not sure how many people were using market orders.. But hey..

**Pull request merged**